utils/generate_grammar_tables.py

```diff
@@ -348,6 +348,91 @@ def generate_operand_kind_table(enums):
     return '\n\n'.join(enum_entries + (table,))
 
 
+def get_extension_list(operands):
+    """Returns extensions as an alphabetically sorted list of strings."""
+    enumerants = sum([item.get('enumerants', []) for item in operands
+                      if item.get('category') in ['ValueEnum']], [])
+
+    extensions = sum([item.get('extensions', []) for item in enumerants
+                      if item.get('extensions')], [])
+
+    return sorted(set(extensions))
+
+
+def get_capability_list(operands):
+    """Returns capabilities as a list of strings in the order of appearance."""
+    enumerants = sum([item.get('enumerants', []) for item in operands
+                      if item.get('kind') in ['Capability']], [])
+    return [item.get('enumerant') for item in enumerants]
+
+
+def generate_extension_enum(operands):
+    """Returns enumeration containing extensions declared in the grammar."""
+    extensions = get_extension_list(operands)
+    return ',\n'.join(['k' + extension for extension in extensions])
+
+
+def generate_extension_to_string_mapping(operands):
+    """Returns mapping function from extensions to corresponding strings."""
+    extensions = get_extension_list(operands)
+    function = 'std::string ExtensionToString(Extension extension) {\n'
+    function += '  switch (extension) {\n'
+    template = '    case Extension::k{extension}:\n' \
+        '      return "{extension}";\n'
+    function += ''.join([template.format(extension=extension)
+                         for extension in extensions])
+    function += '  };\n\n  return "";\n}'
+    return function
+
+
+def generate_string_to_extension_mapping(operands):
+    """Returns mapping function from strings to corresponding extensions."""
+    extensions = get_extension_list(operands)
+    function = 'bool GetExtensionFromString(' \
+        'const std::string& str, Extension* extension) {\n '
+    template = ' if (str == "{extension}")\n' \
```

Comment:
This bakes in the linear search.  If the number of extensions grows large then we might consider doing a regular STL lookup into an unordered map.  Suggest adding a TODO here.

At least this is no worse than the existing algorithm.